<?xml version="1.0" encoding="utf-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from shelf.urdf.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="shelf">
  <link name="world"/>
  <joint name="fixed" type="fixed">
    <parent link="world"/>
    <child link="shelf"/>
    <origin rpy="-0.008 -0.001 -3.085" xyz="0.943 -0.952 0.903"/>
  </joint>
  <link name="shelf">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="5"/>
      <inertia ixx="0.375863761" ixy="0" ixz="0" iyy="0.3397244725" iyz="0" izz="0.375863761"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 -0.1475 -0.4515"/>
      <geometry>
        <box size="0.016 0.295 0.903"/>
      </geometry>
      <material name="shelf_material">
        <color rgba="0.0 1.0 1.0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 -0.1475 -0.4515"/>
      <geometry>
        <box size="0.016 0.295 0.903"/>
      </geometry>
    </collision>
  </link>
  <link name="door">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="5.8"/>
      <inertia ixx="0.4333886585" ixy="0" ixz="0" iyy="0.39146708384" iyz="0" izz="0.4333886585"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0.1475 0"/>
      <geometry>
        <box size="0.016 0.295 0.9"/>
      </geometry>
      <material name="door_material">
        <color rgba="1.0 1.0 1.0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0.1475 0"/>
      <geometry>
        <box size="0.016 0.295 0.9"/>
      </geometry>
      <material name="">
        <contact name="rubber"/>
      </material>
    </collision>
  </link>
  <link name="door_hinge">
    <inertial>
      <mass value="0.01"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
  </link>
  <joint name="door_hinge_joint" type="fixed">
    <parent link="shelf"/>
    <child link="door_hinge"/>
    <dynamics damping="2.0" friction="8.0"/>
    <origin xyz="0 -0.295 0.45"/>
  </joint>
  <link name="axis_link">
    <inertial>
      <mass value="0.01"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
  </link>
  <joint name="axis_joint" type="fixed">
    <parent link="door_hinge"/>
    <child link="axis_link"/>
    <origin xyz="0.004 0.027 -0.150"/>
  </joint>
  <joint name="articulation_joint" type="revolute">
    <parent link="axis_link"/>
    <child link="measured_joint_link"/>
    <dynamics damping="2.0" friction="4.0"/>
    <origin xyz="0 0 0"/>
    <axis xyz="0 0 -1"/>
    <limit effort="100" lower="0" upper="1.57" velocity="10.0"/>
  </joint>
  <joint name="door_link" type="fixed">
    <parent link="measured_joint_link"/>
    <child link="door"/>
    <origin xyz="-0.004 -0.027 0.150"/>
  </joint>
  <link name="handle_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.01 0.01 0.17"/>
      </geometry>
      <material name="handle_material">
        <color rgba="0.0 0.0 0.0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.01 0.01 0.17"/>
      </geometry>
      <material name="">
        <contact name="rubber"/>
      </material>
    </collision>
  </link>
  <link name="handle_support_left">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.056 0.01 0.01"/>
      </geometry>
      <material name="handle_material">
        <color rgba="0.0 0.0 0.0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.056 0.01 0.01"/>
      </geometry>
      <material name="">
        <contact name="steel"/>
      </material>
    </collision>
  </link>
  <link name="handle_support_right">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.056 0.01 0.01"/>
      </geometry>
      <material name="handle_material">
        <color rgba="0.0 0.0 0.0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.056 0.01 0.01"/>
      </geometry>
      <material name="">
        <contact name="steel"/>
      </material>
    </collision>
  </link>
  <joint name="handle_joint" type="fixed">
    <parent link="door"/>
    <child link="handle_link"/>
    <origin rpy="1.57 0 0" xyz="0.064 0.27 -0.15"/>
    <!-- height handle + half the width door, 0.30 minus some offset -->
  </joint>
  <joint name="handle_ref" type="fixed">
    <parent link="handle_link"/>
    <child link="handle_ref"/>
    <origin rpy="0 0 0" xyz="-0.028 0 0"/>
    <!-- height handle + half the width door, 0.30 minus some offset -->
  </joint>
  <link name="handle_ref">
    <inertial>
      <mass value="0.01"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
  </link>
  <joint name="handle_support_left_joint" type="fixed">
    <parent link="handle_link"/>
    <child link="handle_support_left"/>
    <origin rpy="0 0 0" xyz="-0.028 0 -0.08"/>
  </joint>
  <joint name="handle_support_right_joint" type="fixed">
    <parent link="handle_link"/>
    <child link="handle_support_right"/>
    <origin rpy="0 0 0" xyz="-0.028 0 0.08"/>
  </joint>
  <link name="measured_joint_link">
    <inertial>
      <mass value="0.01"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
  </link>
  <link name="measured_joint_link_test"/>
  <joint name="measured_test_handle_joint" type="fixed">
    <parent link="handle_link"/>
    <child link="measured_joint_link_test"/>
    <origin rpy="0 0 0" xyz="-0.06 -0.243 0.0"/>
  </joint>
</robot>

